autotest/gcore/tiff_read.py

```diff
@@ -4573,3 +4573,18 @@ def test_tiff_read_unhandled_codec_unknown_name():
     with gdaltest.error_handler():
         assert gdal.Open("data/gtiff/unsupported_codec_unknown.tif") is None
     assert "missing codec of code 44510" in gdal.GetLastErrorMsg()
+
+
+###############################################################################
+# Test reading a 4 band jxl tiff created before 3.6 where alpha vs undefined
+# channel handling was not explicitely handled (#6393)
+
+def test_tiff_jxl_read_for_files_created_before_6393():
+    gdal.ErrorReset()
+    with gdaltest.error_handler():
+        ds = gdal.Open("data/jxl-rgbi.tif")
```

Comment:
Could you also move it to autotest/gcore/data/gtiff/jxl-rgbi.tif . We try to avoid adding new stuff into autotest/gcore/data/ itself.
Same for be below file, which hasn't been committed